core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java

```diff
@@ -73,6 +73,8 @@ final public class Constants {
             OptionalDouble.class.getName(), Double.class.getName(),
             "java.lang.Class<?>", "class name",
             "java.net.InetSocketAddress", "host:port",
+            "io.quarkus.vertx.http.runtime.TrustedXForwardedProxyCheck.TrustedXForwardedProxyCheckPart",
```

Comment:
I don't think this should be in core.

Reply:
I inspected config generation and to my knowledge this is the only way to do it currently.

Reply:
(btw. it's about config documentation generation - expected value TrustedXForwardedProxyCheckPart won't say too much..)

Reply:
@michalvavrik Sorry, what exactly you are trying to achieve with adding a ref to `io.quarkus.vertx.http.runtime.TrustedXForwardedProxyCheck.TrustedXForwardedProxyCheckPart` in the core annotations processor ? So that a more `typed` config property is generated ? May be it is simpler just to use one f the ConfigItem doc attributes ?

Reply:
Documentation table with configuration options contain property type. The property type in this case would be TrustedXForwardedProxyCheckPart and it would not provide users sufficient information on what should be added. There is no such `@ConfigItem` attribute to my knowledge. Constant above tells the processor how should it translate data type to user-friendly documentation that tells exactly what to do. For example, the option I added tells to use host, host[:port] or CIDR, which are all acceptable value formats.

If in doubt, please lookup property `quarkus.kafka-streams.bootstrap-servers` in generated documentation and be aware, that should `InetSocketAddress` not be there too, you would see `java.net.InetSocketAddress`.

@sberyozkin

Reply:
@michalvavrik Thanks, yes, it just seems like it is constant is added in the wrong place. It is a string but it indeed looks a bit strange to have a vertx package prefixed property. 
I'd not really worry about having just `TrustedXForwardedProxyCheckPart` generated, `ConfigItem` has an attribute allowing to have some extra docs attached, don't recall the name now, and there you can clarify what this type means. Does it sound reasonable ?

Reply:
Might also make sense to have TrustedXForwardedProxyCheckPart a standalone interface but it is up to you if you decide to keep it the current way

Reply:
I dropped the constant. Property description makes it clear what are expected values. Personally, I thought previous state with constant was user friendly, but let's proceed.